valhalla/baldr/conditional_speed_limit.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <cstdint>
+
+namespace valhalla {
+namespace baldr {
+
+/** A combination of `TimeDomain` condition and a speed limit in a single 8b word */
+struct ConditionalSpeedLimit {
+  /**
+   * Provides speed limit value in KPH.
+   * @return Speed limit value in KPH.
+   */
+  uint32_t speed_limit() const {
+    return speed_limit_;
+  }
+
+  /**
+   * Provides the `TimeDomain` value of the condition, that defines when this speed limit is applied.
+   * @return `TimeDomain` value of the condition.
+   */
+  uint64_t condition() const {
+    return condition_;
+  }
+
+  uint64_t condition_ : 56;  // TimeDomain meaningful bits
```

Comment:
can we rename this to `time_domain_` so its clear that that is what this is? also why it is 56 bits, when the `DateRange` object is 54 bits. i see you only use 8 for the speed so you dont need all 10 spare but still dont you want the condition thing to match exactly the timedomain union? i wonder if it would be easier if we use a union here to get at a speed limit that overrides the spare:

```
union conditional_speed_t {
   TimeDomain td_;
   struct speed_limit {
     uint64_t padding_ : 54; // this would pad over the timedomain part
     uint64_t speed_ : 8;
     uint64_t spare_ : 2;
   };
};
```

then we could maybe just throw the to_string function for the condition into the timedomain header/impl and keep this thing tiny?

Reply:
Addressed in 23799605a30ab0ac75d7e4f49f05e03f4b1b2860